docs/src/description/syntax.md

```diff
@@ -1,43 +1,14 @@
-  - `first`: used to access the value of a trace column at the first row of the trace. _It may only be used when defining boundary constraints._
-  - `last`: used to access the value of a trace column at the last row of the trace. _It may only be used when defining boundary constraints._
-- `enf`: used to describe a single [constraint](./constraints.md).
-- `public_inputs`: used to declare the [source section](./structure.md#source-sections) where the [public inputs are declared](./declarations.md). _They may only be referenced when defining boundary constraints._
-- `periodic_columns`: used to declare the [source section](./structure.md#source-sections) where the [periodic columns are declared](./declarations.md). _They may only be referenced when defining integrity constraints._
-- `trace_columns`: used to declare the [source section](./structure.md#source-sections) where the [execution trace is described](./declarations.md).
-  - `main`: used to declare the main execution trace.
-  - `aux`: used to declare the auxiliary execution trace.
-- `integrity_constraints`: used to declare the [source section](./structure.md#source-sections) where the [integrity constraints are described](./constraints.md#integrity_constraints).
-
-## Built-in variables
-
-Built-in variables are identified by the starting character `\$`.
-
-### \$rand
-
-Currently, the only built-in is `\$rand`, which is used to get random values provided by the verifier.
-
-These random values may be accessed by using the indexing operator on `\$rand`. For example, `\$rand[i]` provides the `ith` random value.
-
-Random values may only be accessed within source sections for constraints, i.e. the [`boundary_constraints` section](./constraints.md#boundary-constraints-boundary_constraints) and the [`integrity_constraints` section](./constraints.md#integrity-constraints-integrity_constraints).
+This page specifies the basic syntax and types.
 
 ## Delimiters and special characters
 
 - `:` is used as a delimiter when declaring [source sections](./structure.md#source-sections) and [types](./declarations.md)
 - `.` is used to access a boundary on a trace column, e.g. `a.first` or `a.last`
 - `[` and `]` are used for defining arrays in [type declarations](./declarations.md) and for indexing in [constraint descriptions](./constraints.md)
 - `,` is used as a delimiter for defining arrays in [type declarations](./declarations.md)
-- `\$` is used to indicate a special built-in value. Currently, it is only used with `\$rand` for accessing random values.
+- `\$` is used to access random values or trace columns. To access random values, users can use `\$rand[i]` where `rand` is the identifier that refers to the random values array and `i` is the index of the random value in the array. To access trace columns, users can use `\$main[i]` or `\$aux[i]` where `i` is the index of the column in the main or auxiliary trace respectively.
```

Comment:
Since this isn't general for trace columns, I would adjust it a bit. I think we can also drop the `$rand` example, since there's a bit more complexity there and it's explained elsewhere in detail

"`$` is used to access random values or built-in variables by their identifier. For example, the column at index i in the main execution trace can be accessed by `$main[i]`.